packages/flutter/lib/src/scheduler/binding.dart

```diff
@@ -528,8 +528,14 @@ mixin SchedulerBinding on BindingBase {
         ));
       }
       return _taskQueue.isNotEmpty;
+    } else {
+      addPostFrameCallback((Duration timeStamp) {
```

Comment:
Does this actually fix the problem of tasks not running, though? Adding a postframecallback does not force a frame. So, if currently no frame is running or scheduled, this may never run and tasks will never execute. I believe, there's no guarantee that a frame is running or scheduled at this point?

Reply:
@goderbauer For now I've just added a comment to `SchedulingStrategy` specifying the current behavior, that the strategy is re-evaluated after the next (or current) frame, or when a new task is scheduled. I haven't added any new assertions. I think this may be the least impactful way to go.

Reply:
It is odd to rely on the inner workings of the default scheduling strategy here, because that one could be fully replaced.

I don't have a great immediate idea of what to do here instead, though. Maybe @Hixie has some advice?

Reply:
A different way of looking at it with the documentation change is that rather than relying on the inner workings of the default scheduling strategy, we're defining that when a scheduling strategy returns false, tasks won't be re-evaluated again until "after the next (or current) frame, or until a new task is scheduled."

This shifts the implementation detail dependence so that it's contained within the expectations of the scheduling strategy itself, such that this patch would fix the expectation in the context of the default scheduling strategy. Were a scheduling strategy to be defined that returns false even if there are no frames scheduled, it would then not intrinsically have an expectation that the task/frame would be guaranteed to be scheduled, which I think is okay.

Reply:
Right, using `addPostFrameCallback` doesn't work in the case where, say, the scheduling strategy is "do tasks except on Tuesdays". On a Tuesday, the scheduler strategy would return false, but on Wednesday we wouldn't know to retrigger the task queue.

I wonder if anyone has ever used the scheduling strategy logic. If not, we could just remove it and hard-code our current strategy and then the assumption would be safe and `addPostFrameCallback` would work.

I don't think, with the current API, that adding `addPostFrameCallback` is a good solution because it prevents us from feeling the pain of the broken API (without making the API less broken).